Applications/Spire/Source/Ui/FlatButton.cpp

```diff
@@ -78,185 +67,83 @@ const FlatButton::Style& FlatButton::get_disabled_style() const {
 
 void FlatButton::set_disabled_style(const Style& disabled_style) {
   m_disabled_style = disabled_style;
-  on_style_updated();
-}
-
-connection FlatButton::connect_clicked_signal(
-    const ClickedSignal::slot_type& slot) const {
-  return m_clicked_signal.connect(slot);
-}
-
-void FlatButton::changeEvent(QEvent* event) {
-  if(event->type() == QEvent::FontChange) {
-    m_label->setFont(font());
-  } else if(event->type() == QEvent::EnabledChange) {
-    if(!isEnabled()) {
-      disable_button();
-    } else {
-      enable_button();
-    }
-  }
-}
-
-void FlatButton::focusInEvent(QFocusEvent* event) {
-  if(event->reason() == Qt::ActiveWindowFocusReason) {
-    if(m_last_focus_reason == Qt::MouseFocusReason) {
-      set_hover_stylesheet();
-    } else {
-      set_focus_stylesheet();
-    }
-    return;
-  } else if(event->reason() != Qt::MouseFocusReason) {
-    set_focus_stylesheet();
-  } else {
-    set_hover_stylesheet();
-  }
-  m_last_focus_reason = event->reason();
+  update_style();
 }
 
-void FlatButton::focusOutEvent(QFocusEvent* event) {
-  set_hover_stylesheet();
+void FlatButton::set_font_properties(int pixel_size, int weight) {
+  auto button_font = font();
+  button_font.setPixelSize(pixel_size);
+  button_font.setWeight(weight);
+  setFont(button_font);
 }
 
 void FlatButton::keyPressEvent(QKeyEvent* event) {
-  if(event->key() == Qt::Key_Enter || event->key() == Qt::Key_Return ||
-      event->key() == Qt::Key_Space) {
-    if(m_clickable) {
-      m_clicked_signal();
-    }
-  }
-  QWidget::keyPressEvent(event);
-}
-
-void FlatButton::mousePressEvent(QMouseEvent* event) {
-  event->accept();
-}
-
-void FlatButton::mouseReleaseEvent(QMouseEvent* event) {
-  if(event->button() == Qt::LeftButton && m_clickable) {
-    set_hover_stylesheet();
-    m_last_focus_reason = Qt::MouseFocusReason;
-    if(rect().contains(event->localPos().toPoint())) {
-      m_clicked_signal();
-    }
+  switch(event->key()) {
+    case Qt::Key_Enter:
+    case Qt::Key_Return:
+    case Qt::Key_Space:
+      if(!event->isAutoRepeat()) {
+        Q_EMIT clicked(isChecked());
+      }
+      break;
   }
```

Comment:
Call the parent class' keyPressEvent when not handling a key press.

Reply:
Adding the parent class' keyPressEvent will add the feature that the focus of the button can be changed by arrow keys. Is this feature what we want?

Reply:
At a minimum, any event that is not explicitly handled by the button and part of the specification should bubble up to the parent widget so the parent will have a chance to handle it.